lighthouse-core/gather/gather-runner.js

```diff
@@ -235,14 +237,16 @@ class GatherRunner {
       // Always record devtoolsLog
       .then(_ => driver.beginDevtoolsLog())
       // Begin tracing if requested by config.
-      .then(_ => recordTrace && driver.beginTrace(options.flags))
+      .then(_ => recordTrace && driver.beginTrace(context.flags))
       // Navigate.
-      .then(_ => GatherRunner.loadPage(driver, options))
+      .then(_ => GatherRunner.loadPage(driver, context))
       .then(_ => log.log('statusEnd', status));
 
-    return gatherers.reduce((chain, gatherer) => {
+    return gatherers.reduce((chain, gathererDefn) => {
       return chain.then(_ => {
-        const artifactPromise = Promise.resolve().then(_ => gatherer.pass(options));
+        const gatherer = gathererDefn.instance;
+        context.options = gathererDefn.options || {};
```

Comment:
at the very least let's comment that this is terrible. :)

Reply:
I'll create a new context so it won't be terrible :)